docker-compose.yml

```diff
@@ -0,0 +1,23 @@
+version: "3"
+services:
+  web:
+    build: .
+    ports:
+      - "3000:3000"
+    volumes:
+      - .:/code
+    links:
+      - redis
+      - postgres
+    command: tail -f /dev/null
```

Comment:
What's the reason to `tail -f /dev/null` here? My guess would be that it's just to keep the container alive, so that you can perform other tasks (like actually running the web server) in different consoles, but couldn't we just have this run the web server anyway, and instead we use `docker-compose run` in the `bin-docker/` files? (Inspiration for this comes from here: <https://docs.docker.com/compose/rails/#build-the-project>.)

This may require that we migrate the database first, but I think that should be as simple as adding a line, `RUN rails db:create db:migrate` (or `RUN db:setup`) in the `Dockerfile`.

Reply:
The reason I propose this is so that it should be as simple as `docker-compose up` to get everything running and have a live dev environment.

Reply:
So the tricky thing about `RUN rails db:setup` in the Dockerfile is that it requires the postgres service to be present, so it can't just be part of the setup for the web image, it has to be part of setup for the whole cluster

Reply:
I've looked around a bit, and can't actually find how to get around this issue, which is annoying. I would expect this to be a problem other people had encountered, but it's not obvious from the keywords I searched with. I tried just using `bundler exec rails db:migrate && bundler exec rails s`, but wasn't able to properly test it. (I'm encountering issues where the `/code/` directory of the container is empty, but this only happened after I started making changes. I'm not sure if there's some kind of caching issue going on with how docker's creating the images, or if it's something to do with me using Windows Subsystem for Linux in the same directory, but reverting to your version of the branch doesn't seem to fix it to how it was before — even when I use `docker-compose build` or `docker-compose up --build`.)